train.py

```diff
@@ -1,207 +1,228 @@
-    grp = f.create_group("info")
-    dset = grp.create_dataset("input_size", (1,), dtype='int16')
-    dset[0] = input_size
-    f.close()
-
-    # Create Analysis Plots ###########################################
-    if args.create_plots == 1:
-        print("Plotting results..")
-        import plot
-
-        plot.analyze_pred_vs_actual({   'output_wav':'models/'+name+'/y_test.wav',
-                                            'pred_wav':'models/'+name+'/y_pred.wav', 
-                                            'input_wav':'models/'+name+'/x_test.wav',
-                                            'model_name':name,
-                                            'show_plots':1,
-                                            'path':'models/'+name
-                                        })
-
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("in_file")
-    parser.add_argument("out_file")
-    parser.add_argument("name")
-    parser.add_argument("--training_mode", type=int, default=0)
-    parser.add_argument("--batch_size", type=int, default=4096)
-    parser.add_argument("--max_epochs", type=int, default=1)
-    parser.add_argument("--create_plots", type=int, default=1)
-    parser.add_argument("--input_size", type=int, default=100)
-    parser.add_argument("--split_data", type=int, default=1)
-    args = parser.parse_args()
-    main(args)
\ No newline at end of file
+    x_test_data = x_all[-int(len(x_all) * test_size):]
+    y_test_data = y_all[-int(len(y_all) * test_size):]
+
+    y_test = y_test_data[input_size-1:]
+    indices = np.arange(input_size) + np.arange(len(x_test_data)-input_size+1)[:,np.newaxis]
+    x_test = tf.gather(x_test_data,indices)
+
+    prediction = model.predict(x_test, batch_size=batch_size)
+
+    wavfile.write(f'models/{out_dir}/y_pred.wav', sr, prediction.flatten().astype(np.float32))
+    wavfile.write(f'models/{out_dir}/x_test.wav', sr, x_test_data.flatten().astype(np.float32))
+    wavfile.write(f'models/{out_dir}/y_test.wav', sr, y_test.flatten().astype(np.float32))
+    print("Prediction complete \n")
+
+    # =========================================================================
+    # Save file
+    # -------------------------------------------------------------------------
+    # The  model needs to be saved before the additional h5py metadata can
+    # be added.
+    #
+    # Keras complains about saving legacy H5 files, but the h5py metadata
+    # append is not forward compatible. This suppresses the warning.
+    # Amenable to reviewing h5 metadata usage and dropping if unused or
+    # switching to a lib that does equivalent things for keras files.
+    # =========================================================================
+
+    model_save_path = f'models/{out_dir}/{name}.h5'
+
+    sys.stderr = open(os.devnull, "w")  # silence stderr
+    model.save(model_save_path)
+    sys.stderr = sys.__stderr__  # unsilence stderr
```

Comment:
This is annoying to have to do. There are some less heavyhanded warning suppression methods that didn't work, presumably keras resets those so its warnings won't get suppressed (i know scipy does this). This routes warnings into the void for just this one call.